src/mlpack/methods/reinforcement_learning/environment/cart_pole.hpp

```diff
@@ -0,0 +1,182 @@
+/**
+ * @file cart_pole.hpp
+ * @author Shangtong Zhang
+ *
+ * This file is an implementation of Cart Pole task
+ * https://gym.openai.com/envs/CartPole-v0
+ *
+ * TODO: refactor to OpenAI interface
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+
+#ifndef MLPACK_METHODS_RL_ENVIRONMENT_CART_POLE_HPP
+#define MLPACK_METHODS_RL_ENVIRONMENT_CART_POLE_HPP
+
+#include <mlpack/prereqs.hpp>
+
+namespace mlpack {
+namespace rl {
+
+namespace cart_pole_details {
```

Comment:
I think it would be a good idea to make the parameters part of the CartPole  constructor, that way, someone could use the environment with e.g. a different mass or gravity. I know it's somewhat unlikely that someone is going for it, but at least there is the possibility to do that without modifying the codebase if you just link against mlpack.